command/agent/config/config_test.go

```diff
@@ -2108,3 +2111,103 @@ func TestLoadConfigFile_Bad_Value_Disable_Keep_Alives(t *testing.T) {
 		t.Fatal("should have error, it didn't")
 	}
 }
+
+func TestLoadConfigFile_EnvTemplates(t *testing.T) {
+	cfg, err := LoadConfigFile("./test-fixtures/config-env-templates-simple.hcl")
+
+	if err != nil {
+		t.Fatalf("error loading config file: %s", err)
+	}
+
+	expectedKey := "MY_DATABASE_USER"
+	found := false
+	for _, envTemplate := range cfg.EnvTemplates {
+		if *envTemplate.MapToEnvironmentVariable == expectedKey {
+			found = true
+		}
+	}
+	if !found {
+		t.Fatalf("expected env var name to be populated")
+	}
+}
+
+func TestLoadConfigFile_EnvTemplateComplex(t *testing.T) {
+	cfg, err := LoadConfigFile("./test-fixtures/config-env-templates-complex.hcl")
+
+	if err != nil {
+		t.Fatalf("error loading config file: %s", err)
+	}
+	expectedKeys := []string{
+		"FOO_DATA_LOCK",
+		"FOO_DATA_PASSWORD",
+		"FOO_DATA_USER",
+	}
+
+	envExists := func(key string) bool {
+		for _, envTmpl := range cfg.EnvTemplates {
+			if *envTmpl.MapToEnvironmentVariable == key {
+				return true
+			}
+		}
+		return false
+	}
+
+	for _, expected := range expectedKeys {
+		if !envExists(expected) {
+			t.Fatalf("expected env var %s", expected)
+		}
+	}
+}
+
+func TestLoadConfigFile_EnvTemplateNoName(t *testing.T) {
+	_, err := LoadConfigFile("./test-fixtures/config-env-templates-no-name.hcl")
+	if err == nil {
+		t.Fatalf("expected error")
+	}
+}
+
+func TestLoadConfigFile_ExecSimple(t *testing.T) {
+	cfg, err := LoadConfigFile("./test-fixtures/config-env-templates-simple.hcl")
+
+	if err != nil {
+		t.Fatalf("error loading config file: %s", err)
+	}
+
+	if cfg.Exec == nil {
+		t.Fatal("expected exec config to be parsed")
+	}
+
+	expectedCmd := []string{"/path/to/my/app", "arg1", "arg2"}
+	if !slices.Equal(cfg.Exec.Command, expectedCmd) {
+		t.Fatal("exec.command does not have expected value")
+	}
+
+	// check defaults
+	if cfg.Exec.RestartOnSecretChanges != "always" {
+		t.Fatalf("expected cfg.Exec.RestartOnSecretChanges to be 'always', got '%s'", cfg.Exec.RestartOnSecretChanges)
+	}
+
+	if cfg.Exec.RestartKillSignal != os.Interrupt {
+		t.Fatalf("expected cfg.Exec.RestartKillSignal to be 'os.Interrupt', got '%s'", cfg.Exec.RestartKillSignal)
+	}
+}
+
+func TestLoadConfigFile_ExecComplex(t *testing.T) {
+	cfg, err := LoadConfigFile("./test-fixtures/config-env-templates-complex.hcl")
+
+	if err != nil {
+		t.Fatalf("error loading config file: %s", err)
+	}
+
+	if !slices.Equal(cfg.Exec.Command, []string{"env"}) {
+		t.Fatal("exec.command does not have expected value")
+	}
+
+	if cfg.Exec.RestartOnSecretChanges != "never" {
+		t.Fatalf("expected cfg.Exec.RestartOnSecretChanges to be 'never', got %q", cfg.Exec.RestartOnSecretChanges)
+	}
+
+	if cfg.Exec.RestartKillSignal != syscall.SIGTERM {
```

Comment:
Could we add a test with an invalid signal as well, something like "SIG-DOES-NOT-EXIST"